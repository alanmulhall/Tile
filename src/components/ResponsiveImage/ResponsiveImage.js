import styled from 'styled-components';

const ResponsiveImageStyled = styled.img`
  height: 100%;
  transition: all 0.2s ease-in-out;
  object-fit: scale-down;
  width: 100%;
`;

export default ResponsiveImageStyled;
