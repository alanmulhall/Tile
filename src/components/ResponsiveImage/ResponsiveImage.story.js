import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import styled from 'styled-components';
import withTests from '../../utils/withTests';
import ResponsiveImage from './ResponsiveImage';
import logo from '../../images/logo.png';

const Container = styled.div`
  align-items: center;
  display: flex;
`;

const SmallImage = styled.figure`
  height: 50px;
`;

const MediumImage = styled.figure`
  height: 100px;
`;

const LargeImage = styled.figure`
  height: 150px;
`;

const ResponsiveImageStory = () =>
  storiesOf('ResponsiveImage', module)
    .addDecorator(centered)
    .addDecorator(withTests('ResponsiveImage'))
    .add('Default', () => (
      <Container>
        <SmallImage>
          <ResponsiveImage src={logo} alt="7" />
        </SmallImage>
        <MediumImage>
          <ResponsiveImage src={logo} alt="7" />
        </MediumImage>
        <LargeImage>
          <ResponsiveImage src={logo} alt="7" />
        </LargeImage>
      </Container>
    ));

export default ResponsiveImageStory;
