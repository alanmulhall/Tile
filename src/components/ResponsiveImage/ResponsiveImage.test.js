import React from 'react';
import 'jest-styled-components';
import renderer from '../../utils/rendererThemeProviderWrapper.js';
import ResponsiveImage from './ResponsiveImage';

describe('<ResponsiveImage />', () => {
  it('renders successfullly', () => {
    const tree = renderer(<ResponsiveImage />);
    expect(tree).toMatchSnapshot();
  });
});
