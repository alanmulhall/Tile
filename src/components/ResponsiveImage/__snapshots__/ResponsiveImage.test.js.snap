// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveImage /> renders successfullly 1`] = `
.c0 {
  height: 100%;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  object-fit: scale-down;
  width: 100%;
}

<img
  className="c0"
/>
`;
